src/CatalogService/src/Application/Product/Commands/AddProduct/AddProductCommand.cs

```diff
@@ -1,10 +1,12 @@
-﻿using Catalog.Abstractions;
+﻿using AutoMapper;
+using Catalog.Abstractions;
 using CatalogService.Application.Common.Interfaces;
+using CatalogService.Application.Common.Models;
 using MediatR;
 
 namespace CatalogService.Application.Product.Commands.AddProduct;
 
-public class AddProductCommand : IRequest<int>
+public class AddProductCommand : IRequest<ProductDto>

```

Comment:
Returning the model as a result of the command violates CQS. It might be ok in some rare cases, but it's hard to support in future. Please, send another request to the mediator to get a needed model